File: src/easy/MoveZeroes.java
package easy;


public class MoveZeroes {
    public void moveZeroes(int[] nums) {
        if (nums == null || nums.length < 2) {
            return;
        }
        int j = 0;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] != 0) {
                nums[j] = nums[i];
                j++;
            }
        }
        while (j < nums.length) {
            nums[j++] = 0;
        }
    }
}

File: csharp/LeetCode/Easy/MoveZeroes.cs
namespace LeetCode.Easy
{
    public partial class Solution
    {
        public void MoveZeroes(int[] nums)
        {
            if (nums == null)
            {
                return;
            }

            var left = 0;
            for (var i = 0; i < nums.Length; i++)
            {
                if (nums[i] != 0)
                {
                    if (left != i)
                    {
                        nums[left] = nums[i];
                        nums[i] = 0;
                    }

                    left++;
                }
            }
        }
    }
}
